lib/internal/modules/cjs/loader.js

```diff
@@ -85,6 +85,17 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  if (
+    err.message.startsWith('SyntaxError: Unexpected token export') ||
+    (/(?:import).*\n{1}\s*\^{9}/mi).test(err.stack)) {
```

Comment:
Didn't you say you were going to try `err.stack.split('\n')[index]` for this? Or did that not work out?

Reply:
Any update on this?

Reply:
Yes, working with splitting the stack by newline and iterating for the line `^^^^^^^^` and getting the previous line, checking if it matches `/import/` should do the trick -- I can guarantee somehow 100% that's precisely detecting it.

Reply:
reworked this a little bit